Sub/Src/ComputerVision/vision.py

```diff
@@ -1,99 +1,188 @@
 
-    def run(self):
-        '''
-        The run loop for performing the vision operations and sending
-        the video video stream.
+        #--CAMERA INSTANCE--#
+        front_camera_index = int(self.param_serv.get_param("Vision/front_camera_index"))
+        self.capture = cv2.VideoCapture(front_camera_index)
 
-        Parameters:
-            N/A
-        Returns:
-            N/A
-        '''
-        video_stream_test = False
-        while(self.run_thread):
+        # Maximum image size the Tegra allows without timeout
+        self.capture.set(3, 450)
+        self.capture.set(4, 450)
 
-            #Attempt
-            if(video_stream_test):
-                self.image_publisher.send_image("test", self.test_image)
-            else:
-                successful_capture, video_frame = self.video_capture.read()
+        # Neural Network Loaded from instance in darknet module
+        darknet_path = self.param_serv.get_param("Vision/yolo/darknet_path")
+        config_file = self.param_serv.get_param("Vision/yolo/config_file")
+        weights_file = self.param_serv.get_param("Vision/yolo/weights_file")
+        metadata_file = self.param_serv.get_param("Vision/yolo/metadata_file")
+        config_file_path = (os.path.join(darknet_path, config_file)).encode()
+        weights_file_path = (os.path.join(darknet_path, weights_file)).encode()
+        metadata_file_path = (os.path.join(darknet_path, metadata_file)).encode()
+        print(config_file_path, weights_file_path)
+        self.net  = load_net(config_file_path,
+                       weights_file_path,
+                       0)
 
-                self.image_publisher.send_image("Vid", video_frame)
+        self.meta = load_meta(metadata_file_path)
 
+        self.distance_calculator = 5
 
-    def kill(self):
+    def run(self):
         '''
-        Safe clean up when killing this thread.
-
-        Paramaeters:
-            N/A
-        Returns:
-            N/A
+        The run loop reads image data from the webcam, processes it through Yolo, and
+        then encodes it into a byte stream and encapsulation frame to be sent over the socket.
         '''
-        print("[INFO]: Killing vision thread. Releasing Resources.")
-        self.video_capture.release()
+        while(True):
+            # Capture frame-by-frame
+            ret, byte_frame = self.capture.read()
+
+            # Operations on the frame
+            '''
+            Yolo: Operations on Frame For Yolo
+            '''
+            if ret:
+                r = detect(self.net, self.meta, byte_frame)
+                #Savind detetion to class attribute
+                self.yolo_detections = r
+
+                #Draw detections in photo
+                for i in r:
+                    x, y, w, h = i[2][0], i[2][1], i[2][2], i[2][3]
+                    self.distance_calculator = distance_calculator(i, x, y, w, h)
+                    self.distance_calculator.set_matrices()
```

Comment:
Like I mentioned earlier, this function does many repetitive things that can be done once. I'm not if it would affect performance at all, just thought I would mention it.